src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java

```diff
@@ -61,9 +61,66 @@ public static ConnectablePositionInfos toMapConnectablePosition(Identifiable<?>
         return builder.build();
     }
 
-    public static CurrentLimitsData toMapDataCurrentLimits(Branch<?> branch, Branch.Side side) {
-        Optional<CurrentLimits> limits = side == Branch.Side.ONE ? branch.getCurrentLimits1() : branch.getCurrentLimits2();
-        return limits.map(l -> toMapDataCurrentLimits(l)).orElse(null);
+    public static Optional<ActivePowerControlInfos> toActivePowerControl(Identifiable<?> identifiable) {
+        ActivePowerControlInfos.ActivePowerControlInfosBuilder builder = ActivePowerControlInfos.builder();
+        var activePowerControl = identifiable.getExtension(ActivePowerControl.class);
+        if (activePowerControl != null) {
+            builder.activePowerControlOn(activePowerControl.isParticipate());
+            builder.droop(activePowerControl.getDroop());
+            return Optional.of(builder.build());
+        }
+        return Optional.empty();
+    }
+
+    public static String toBranchStatus(Branch<?> branch) {
+        var branchStatus = branch.getExtension(BranchStatus.class);
+        return branchStatus == null ? null : branchStatus.getStatus().name();
+    }
+
+    public static Optional<GeneratorShortCircuitInfos> toGeneratorShortCircuit(Generator generator) {
+        GeneratorShortCircuit generatorShortCircuit = generator.getExtension(GeneratorShortCircuit.class);
+        if (generatorShortCircuit != null) {
+            GeneratorShortCircuitInfos.GeneratorShortCircuitInfosBuilder builder = GeneratorShortCircuitInfos.builder();
+            builder.transientReactance(generatorShortCircuit.getDirectTransX());
+            builder.stepUpTransformerReactance(generatorShortCircuit.getStepUpTransformerX());
+            return Optional.of(builder.build());
+        }
```

Comment:
this `if` block is not cover by a test

Reply:
I added this extension in existing test to cover it